main.js

```diff
@@ -0,0 +1,224 @@
+//defined function to check if each row was equal to a winning state
+function arraysEqual(arr1, arr2) {
+  if (arr1.length !== arr2.length) {
+    return false;
+  }
+
+  //Loop through arr1 and see if each item is equal to arr2
+  for (let i = 0; i < arr1.length; i++) {
+    if (arr1[i] !== arr2[i]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+//Since printBoard() inserts new HTML elements each time, an addListeners() function is defined to easily add listeners back to the new elements
+function addListeners() {
+  var pegs = document.querySelectorAll('.peg')
+
+  pegs.forEach(function(peg) {
+    peg.addEventListener('click', handlePegClick)
+  })
+}
+
+var game = {
+  board: [
+    [5, 4, 3, 2, 1],
+    [],
+    [],
+    [],
+    []
+  ],
+  winningStack: [5, 4, 3, 2, 1],
+  visualBoard: [],
+  firstPegSelect: false,
+  firstPegVal: 0,
+  printBoard: function() {
+    newBoard = []
+    //Loop thorugh each row on the board
+    for(let i = 0; i < this.board.length; i++) {
+      //map it to a new array which turns the numbers to strings then joins them with a space
+      const rowString = this.board[i].map(num => num.toString()).join(' ');
+      //push p element to newBoard and set id to i and class to pet
+      newBoard.push(`<p class='peg' id=${i}>--- ${rowString}</p>`);
+      //set the visual board to the new board outside the for loop
+      this.visualBoard = newBoard;
+    };
+    //update the HTML to reflect the new board
+    document.querySelector(".towers-container").innerHTML = this.visualBoard.join('')
+    addListeners()
+  },
+  //function to log a message with the new board
+  displayMove: function() {
+    document.querySelector('.message').innerHTML = 'That move was successful, the board is now:'
+    this.printBoard()
+  },
+  resetBoard: function() {
+    //loop through the board and add an empty array for each item based on the current length
+    for(let i = 0; i < this.board.length; i++) {
+     this.board[i] = [] 
+    }
+    //set the first peg equal to the winning stack
+    this.board[0] = this.winningStack
+    message('Board reset.')
+    this.printBoard()
+  },
+  //checks if the board is in a winning state
+  checkWinner: function() {
+    let winner = false;
+    //loop through each array in the board 2D array. Skip the first array (set i = 1) because you cannot win if they are lined up on the starting peg
+    for(let i = 1; i < this.board.length; i++) {
+      //if its equal to the winning stack, reset the game, otherwise run display move
+      if(arraysEqual(this.board[i], this.winningStack)) {
+      winner = true;
+      this.printBoard()
+      message("you win! resetting...");
+      setTimeout(() => this.resetBoard(), 2000)
+    }
+    }
+  if(winner === false) {
```

Comment:
review conditions - you can write this as if (winner) because conditions evaluate to boolean and since its already a boolean you don't have to evaluate it